src/librustc_typeck/diagnostics.rs

```diff
@@ -2576,6 +2711,28 @@ to change this.
 [RFC 953]: https://github.com/rust-lang/rfcs/pull/953
 "##,
 
+E0369: r##"
+A binary operation was attempted on a type which doesn't support it.
+Erroneous code example:
+
+```
+let x = 12f32; // error: binary operation `<<` cannot be applied to
+               //        type `f32`
+
+x << 2;
+```
+
+To fix this error, please check this type implements this binary operation.
+Example:
```

Comment:
It is also possible to overload most operators for your own type by implementing traits from `std::ops`